mifos-passcode-cmp/src/commonMain/kotlin/auth/passcode/screen/PasscodeScreen.kt

```diff
@@ -158,16 +165,26 @@ fun PasscodeScreen(
                 },
                 hasPassCode = state.isPasscodeAlreadySet
             )
+
+            Spacer(modifier = Modifier.height(20.dp))
+
+            PasscodeLengthChangeButton(
+                onPasscodeLengthChange = {
+                    passcodeSaver.changePasscodeChangeSwitchState()
+                },
```

Comment:
The `PasscodeLengthChangeButton`  onPasscodeLengthChange sends an Int (4 or 6), but PasscodeSaver still handles switching internally.

Update PasscodeSaver:
```
fun updatePasscodeLength(length: Int) {
    updateState {
        copy(
            passcodeLength = length,
            passcodeLengthSwitchChecked = length == 6
        )
    }
}

```
Update Composable onPasscodeLengthChange:
```
onPasscodeLengthChange = { length -> passcodeSaver.updatePasscodeLength(length) }
```